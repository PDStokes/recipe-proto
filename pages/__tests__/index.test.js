import axios from 'axios';
import parser from '../../plugins/siteParser';

const testUrlList = [
    'https://thewoksoflife.com/garlic-chives-pork-cang-ying-tou/',
    'https://www.mylatinatable.com/best-tacos-al-pastor-recipe/',
    'https://lovelylittlekitchen.com/best-ever-pumpkin-muffins/',
    'https://www.cleanandscentsible.com/chocolate-rice-krispie-gingerbread-men-pops/',
];

describe('Recipe URL snapshots', () => {
    testUrlList.forEach(url => {
        test(`${url}:`, async () => {
            const response = await axios.get(url);
            const parsedData = parser.parse(response.data);
            expect(parsedData).toMatchSnapshot();
        });
    });
});
