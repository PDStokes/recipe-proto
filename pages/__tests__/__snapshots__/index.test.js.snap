// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Recipe URL snapshots https://lovelylittlekitchen.com/best-ever-pumpkin-muffins/: 1`] = `
Object {
  "directions": Array [
    Object {
      "header": true,
      "text": "Instructions",
    },
    Object {
      "header": false,
      "text": "Preheat the oven to 375 degrees and place 12 paper liners into each well of your standard size muffin baking pan.",
    },
    Object {
      "header": false,
      "text": "Measure out the flour, sugars, baking soda, salt and spices in a medium bowl and whisk together. Set aside.",
    },
    Object {
      "header": false,
      "text": "In another bowl, whisk together the eggs, pumpkin puree, coconut oil and vanilla extract.",
    },
    Object {
      "header": false,
      "text": "Pour the wet ingredients into the dry ingredients and stir together. Do not over mix, just stir until everything is incorporated into the batter.",
    },
    Object {
      "header": false,
      "text": "It is helpful to use a large scoop (like an ice cream scoop) to evenly distribute the batter into each well. They will be nearly full. This will help give your muffins a nice puffy dome.",
    },
    Object {
      "header": false,
      "text": "Bake your muffins for 20-22 minutes or until a toothpick inserted into the center of a muffin comes out clean.",
    },
  ],
  "ingredients": Array [
    Object {
      "header": true,
      "text": "Ingredients",
    },
    Object {
      "header": false,
      "text": "1 3/4 cups all purpose flour",
    },
    Object {
      "header": false,
      "text": "1 cup sugar",
    },
    Object {
      "header": false,
      "text": "1/2 cup dark brown sugar",
    },
    Object {
      "header": false,
      "text": "1 teaspoon baking soda",
    },
    Object {
      "header": false,
      "text": "1/2 teaspoon salt",
    },
    Object {
      "header": false,
      "text": "2 teaspoons cinnamon",
    },
    Object {
      "header": false,
      "text": "1/4 teaspoon ground cloves",
    },
    Object {
      "header": false,
      "text": "1/4 teaspoon nutmeg",
    },
    Object {
      "header": false,
      "text": "2 eggs",
    },
    Object {
      "header": false,
      "text": "1 15 ounce can pure pumpkin puree",
    },
    Object {
      "header": false,
      "text": "1/2 cup coconut oil, melted",
    },
    Object {
      "header": false,
      "text": "1 teaspoon vanilla extract.",
    },
  ],
  "title": "Best Ever Pumpkin Muffins",
}
`;

exports[`Recipe URL snapshots https://thewoksoflife.com/garlic-chives-pork-cang-ying-tou/: 1`] = `
Object {
  "directions": Array [
    Object {
      "header": false,
      "text": "Preheat your wok over high heat until it starts to smoke slightly. Turn down the heat, and let the wok cool down a bit before adding 1 tablespoon of the oil. (This step of superheating the wok before adding oil prevents the meat from sticking to the wok. This method works well for pan-frying anything--fish included.)",
    },
    Object {
      "header": false,
      "text": "Once the oil is heated, turn the heat down to medium, and add the ground pork. Cook it until the meat is lightly browned. Add in the Shaoxing wine, ½ tablespoon light soy sauce, and the sugar. Mix well, turn off the heat, and transfer the cooked meat to a bowl.",
    },
    Object {
      "header": false,
      "text": "Now turn the heat down to low, and add the last 2 tablespoons of oil and the ginger. Cook it for a minute, then add the garlic and fermented black beans. Cook everything for a minute or two, taking care to avoid burning.",
    },
    Object {
      "header": false,
      "text": "Next, add peppers, and turn the heat up to high. Stir fry everything together, and let it cook for a minute uncovered. Then, add in the cooked pork and garlic chives (if you use scapes, you will need to extend the cooking time at the end for 1-2 minutes with the wok lid on) and stir-fry.",
    },
    Object {
      "header": false,
      "text": "Now it’s time to add in the remaining light soy sauce (1½ tablespoons) and the sesame oil. Stir-fry to combine. At this point, you can salt to taste, but the fermented black beans are quite salty--I found that I didn’t need to add any salt. But be fast about it, because the garlic chives cook quickly.",
    },
    Object {
      "header": false,
      "text": "Transfer to a dish and serve with steamed rice.",
    },
  ],
  "ingredients": Array [
    Object {
      "header": false,
      "text": "3 tablespoons oil (divided)",
    },
    Object {
      "header": false,
      "text": "10 ounces ground pork (285 g; can also use ground chicken, turkey, or beef)",
    },
    Object {
      "header": false,
      "text": "2 teaspoons Shaoxing wine",
    },
    Object {
      "header": false,
      "text": "2 tablespoons light soy sauce (divided)",
    },
    Object {
      "header": false,
      "text": "¼ teaspoon sugar",
    },
    Object {
      "header": false,
      "text": "½ tablespoon ginger (finely chopped)",
    },
    Object {
      "header": false,
      "text": "1 tablespoon garlic (finely chopped)",
    },
    Object {
      "header": false,
      "text": "¼ cup fermented black beans (35 grams, rinsed and drained)",
    },
    Object {
      "header": false,
      "text": "½ cup red long hot peppers (or red bell peppers, finely diced)",
    },
    Object {
      "header": false,
      "text": "12 ounces garlic chives (cut into ⅓-inch pieces)",
    },
    Object {
      "header": false,
      "text": "½ teaspoon sesame oil",
    },
  ],
  "title": "Garlic Chives with Pork (Cang Ying Tou - 苍蝇头)",
}
`;

exports[`Recipe URL snapshots https://www.cleanandscentsible.com/chocolate-rice-krispie-gingerbread-men-pops/: 1`] = `
Object {
  "directions": Array [
    Object {
      "header": false,
      "text": "Over low heat melt the butter. Add in the marshmallows and stir to melt. Combine with cocoa powder and remove from heat.",
    },
    Object {
      "header": false,
      "text": "Stir in the Rice Krispies until combined.",
    },
    Object {
      "header": false,
      "text": "Grease the inside of a gingerbread man cookie cutter {as well as your hands!} and place on waxed paper or parchment paper. Press the Rice Krispie mixture firmly into the mold. If you are making pops, you will need to make the gingerbread men a little thicker {about 1 inch} to hold the stick in place. If you are just making the gingerbread men with no stick, they do not need to be as thick.",
    },
    Object {
      "header": false,
      "text": "Remove the Rice Krispie treat from the cookie cutter and place on waxed paper or parchment paper. Twist the stick in at the bottom. Keep firm pressure on the gingerbread man so he stays together. You may need to just re-form the shape a bit once the stick has been inserted. Press firmly together.",
    },
    Object {
      "header": false,
      "text": "Decorate with cookie icing and candy as desired.",
    },
  ],
  "ingredients": Array [
    Object {
      "header": false,
      "text": "¼ cup butter + more for greasing",
    },
    Object {
      "header": false,
      "text": "6 cups marshmallows",
    },
    Object {
      "header": false,
      "text": "1 tbsp. cocoa powder",
    },
    Object {
      "header": false,
      "text": "5½ cups chocolate Rice Krispies",
    },
    Object {
      "header": false,
      "text": "white cookie icing",
    },
    Object {
      "header": false,
      "text": "M&Ms or other candy for buttons",
    },
    Object {
      "header": false,
      "text": "paper straw {if desired} or popsicle sticks",
    },
  ],
  "title": "Chocolate Rice Krispie Gingerbread Men Pops",
}
`;

exports[`Recipe URL snapshots https://www.mylatinatable.com/best-tacos-al-pastor-recipe/: 1`] = `
Object {
  "directions": Array [
    Object {
      "header": true,
      "text": "For the Al Pastor Meat/Marinade",
    },
    Object {
      "header": false,
      "text": "Blend all of the ingredients together until completely combined.",
    },
    Object {
      "header": false,
      "text": "Strain the liquid over a bowl to end up with a smooth marinade.",
    },
    Object {
      "header": false,
      "text": "Add one layer of meat in a large bowl or baking dish and cover with the marinade, and then repeat in layers until all of the meat and marinade have been used.",
    },
    Object {
      "header": false,
      "text": "Cover with aluminum foil and refrigerate for at least 4 hours (overnight is best)",
    },
    Object {
      "header": false,
      "text": "Remove from the refrigerator and grill the meat until cooked through.",
    },
    Object {
      "header": false,
      "text": "Remove from grill and cut into thin slices to serve on the tacos.",
    },
    Object {
      "header": true,
      "text": "For the Roasted Tomatillo Chipotle Salsa",
    },
    Object {
      "header": false,
      "text": "Roast the tomatillo, onion, and garlic in the oven on baking sheet at 450 degrees for 8-10 minutes, remove from oven.",
    },
    Object {
      "header": false,
      "text": "Blend with the rest of the ingredients except the salt and pepper until well combined.",
    },
    Object {
      "header": false,
      "text": "Put in a bowl and season with salt and pepper to taste.",
    },
    Object {
      "header": true,
      "text": "For the Tacos",
    },
    Object {
      "header": false,
      "text": "Heat the tortillas with a little bit of oil.",
    },
    Object {
      "header": false,
      "text": "Add the meat, the pineapple, the salsa, the lime juice, the cilantro and onions, and enjoy!",
    },
  ],
  "ingredients": Array [
    Object {
      "header": true,
      "text": "For the Al Pastor Meat/Marinade",
    },
    Object {
      "header": false,
      "text": "2-3 pound Pork Loin cut into thin slices",
    },
    Object {
      "header": false,
      "text": "4 Guajillo Chile boiled, de-seeded, and cleaned.",
    },
    Object {
      "header": false,
      "text": "1 Ancho Chile boiled, de-seeded, and cleaned.",
    },
    Object {
      "header": false,
      "text": "4 cloves of garlic",
    },
    Object {
      "header": false,
      "text": "1/2 onion",
    },
    Object {
      "header": false,
      "text": "1/2 cup of orange juice",
    },
    Object {
      "header": false,
      "text": "3/4 cup of pineapple juice fresh is better",
    },
    Object {
      "header": false,
      "text": "1 Tablespoon of Oregano",
    },
    Object {
      "header": false,
      "text": "1 Tablespoon of Thyme",
    },
    Object {
      "header": false,
      "text": "1 Tablespoon of Pepper",
    },
    Object {
      "header": false,
      "text": "1 Tablespoon of Cumin",
    },
    Object {
      "header": false,
      "text": "1-2 Tablespoons of Paprika",
    },
    Object {
      "header": false,
      "text": "1-2 Tablespoons of Salt",
    },
    Object {
      "header": false,
      "text": "3 Cloves",
    },
    Object {
      "header": true,
      "text": "For the Tacos",
    },
    Object {
      "header": false,
      "text": "15 Corn Tortillas",
    },
    Object {
      "header": false,
      "text": "Chopped cilantro",
    },
    Object {
      "header": false,
      "text": "Chopped Onions",
    },
    Object {
      "header": false,
      "text": "Lime juice",
    },
    Object {
      "header": false,
      "text": "Salt",
    },
    Object {
      "header": false,
      "text": "Grilled pineapple",
    },
    Object {
      "header": false,
      "text": "Tacos al pastor meat",
    },
    Object {
      "header": true,
      "text": "For the Roasted Tomatillo Chipotle Sauce",
    },
    Object {
      "header": false,
      "text": "3 Tomatillos",
    },
    Object {
      "header": false,
      "text": "1/4 Onion",
    },
    Object {
      "header": false,
      "text": "2 small cloves of garlic",
    },
    Object {
      "header": false,
      "text": "1-2 chipotles depending on level of spiciness you want",
    },
    Object {
      "header": false,
      "text": "3 Tablespoons of Pineapple juice",
    },
    Object {
      "header": false,
      "text": "Salt and Pepper to taste",
    },
  ],
  "title": "The Best Tacos Al Pastor Recipe",
}
`;
